### DISCUSSION TITLE
Gravis Ultrasound questions, bugs and possible fixes.

### DISCUSSION THREAD

**Discussion by Fenix770** [86Box/86Box#1469] [Ideas]
Hello!

I found a file that seems to be the source code of Gravis, SBOS software. I know this software is to emulate the Sound Blaster 1.x only. I don't know if useful for the emulated GUS classic and MAX which are emulated on 86box. Because I'm not a programmer and a similar software IwSBOS exist but is for the latter GUS PnP.

Here is the file:
[amd960214.tar.gz](https://github.com/86Box/86Box/files/6587956/amd960214.tar.gz)

Some of the old bugs are still present on GUS is:

1.The SETUP.EXE utility located on the C:\ULTRASND directory will fail and freeze the VM during the test of SB and Roland  emulation if the CPU is in V86 mode (for example if EMM386, QEMM, 386MAX or any other 386 memory manager is installed.)
The test works if no EMS/UMB memory manager is present. 

2. The SBOS utility work,s but, without any 386 memory manager and most software detects it as ADLIB sound card and not Sound Blaster (Example: Starfire DOS game)

3. MEGA-EM v2.07 works fine with any 386 memory manager (tested TIM 2 under DOS). But this emulator is different that SBOS. SBOS emulates the SB 1.x card for music and sound. MEGA-EM emulates the Roland sound card for music and SB 2.0 for sound only. Some games such as PoP 1.3 supports the Roland Sound card **but** the GUS Roland emulation is worse than the SBOS.

4. MEGA-EM 3.x versions, launched in 1995 combines both SBOS and MEGA-EM utilities in only one software and add GUS MAX support. This does not work on any emulator because requires a VCPI compliant EMM driver (386 memory manager, but the SBOS code crash the system on all of them, if EMM386 or any other manager is present)

5. There are other strange bugs on Windows 3.1 (without emulation) some games not detects correctly the card (Example: The even more incredible machine and some other SIERRA games) and others have a 1 second delay after play a music file. (Example PC globe CD-ROM when playing an anthem).

**Comment by volkertb** [comment]
As well-intentioned as this is, be careful with sharing source code that doesn't have a clear legal status.

Was this officially open-sourced by the copyright owners?

Until there is clarity about that, I'm going to refrain from downloading it, and I advise others not to download it either.
